docs/04-operation-guides/assets/sidecar-analysis.sh

```diff
@@ -0,0 +1,72 @@
+#!/bin/bash
+
+target_namespace=$1
+
+log_pods_with () {
+    namespace=$1
+    label=$2
+
+    if [ "$label" != "" ]; then
+        pods_out_of_istio=$(kubectl get pod -l $label -n $namespace -o jsonpath='{.items[*].metadata.name}')
+        for pod in $pods_out_of_istio
+        do
+            if [ "$target_namespace" == "" ]; then
+                echo "    - $namespace/$pod"
+            else
+                echo "  - $pod"
+            fi
+        done
+    else
+        pods_out_of_istio=$(kubectl get pod -n $namespace -o jsonpath='{.items[*].metadata.name}')
+        for pod in $pods_out_of_istio
+        do
+            if [ "$target_namespace" == "" ]; then
+                echo "    - $namespace/$pod"
+            else
+                echo "  - $pod"
+            fi
+        done
+    fi
+}
```

Comment:
```suggestion
#!/bin/bash

set -eou pipefail

target_namespace="${1:-}"

log_pods_with () {
    namespace=$1
    label=$2

    if [ "$label" != "" ]; then
        cmd="kubectl get pod -l $label -n $namespace -o jsonpath='{.items[*].metadata.name}'"
    else
        cmd="kubectl get pod -n $namespace -o jsonpath='{.items[*].metadata.name}'"
    fi

    pods_out_of_istio=$(eval $cmd)
    for pod in $pods_out_of_istio
    do
        if [ "$target_namespace" == "" ]; then
            echo "    - $namespace/$pod"
        else
            echo "  - $pod"
        fi
    done
}
```

Reply:
another minor think would be to utilise bash `-z` and `-n` comparison operators for the `if`'s: https://tldp.org/LDP/abs/html/comparison-ops.html